plugin/kubernetes/kubernetes.go

```diff
@@ -143,7 +143,7 @@ func (k *Kubernetes) Lookup(state request.Request, name string, typ uint16) (*dn
 
 // IsNameError implements the ServiceBackend interface.
 func (k *Kubernetes) IsNameError(err error) bool {
-	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest
+	return err == errNoItems || err == errNsNotExposed || err == errInvalidRequest || err == errPodsDisabled
```

Comment:
This smells like a hack. errPodsDisabled should be used in a general IsNameError function.

Reply:
not following...

Reply:
What has podDisabled to do with nxdomain? errInvalidRequest is already border line

Reply:
If the podMode is set to podsDisabled (default setting in k8s plugin), there we don't produce A records for pods (e.g. 1-2-3-4.ns.pod.cluster.local)...

If some one tries to query a pod in this mode, then there will be no `A` record for it. `NXDOMAIN` seems like the correct answer here.